bridge-token/src/lib.rs

```diff
@@ -134,3 +149,49 @@ impl FungibleTokenMetadataProvider for BridgeToken {
 }
 
 admin_controlled::impl_admin_controlled!(BridgeToken, paused);
+
+// Migration
+
+#[cfg(feature = "migrate_icon")]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct BridgeTokenV0 {
+    controller: AccountId,
+    token: FungibleToken,
+    name: String,
+    symbol: String,
+    reference: String,
+    reference_hash: Base64VecU8,
+    decimals: u8,
+    paused: Mask,
+}
+
+#[cfg(feature = "migrate_icon")]
+impl From<BridgeTokenV0> for BridgeToken {
+    fn from(obj: BridgeTokenV0) -> Self {
+        Self {
+            controller: obj.controller,
+            token: obj.token,
+            name: obj.name,
+            symbol: obj.symbol,
+            reference: obj.reference,
+            reference_hash: obj.reference_hash,
+            decimals: obj.decimals,
+            paused: obj.paused,
+            icon: None,
+        }
+    }
+}
+
+#[cfg(feature = "migrate_icon")]
+#[near_bindgen]
+impl BridgeToken {
+    /// Adding icon as suggested here: https://nomicon.io/Standards/FungibleToken/Metadata.html
+    /// This function can only be called from the factory or from the contract itself.
+    #[init(ignore_state)]
+    pub fn migrate_nep_148_add_icon() -> Self {
+        let old_state: BridgeTokenV0 = env::state_read().unwrap();
```

Comment:
Why do we `unwrap`? I think it's better to be explicit putting something like:
`env::state_read().expect("The state doesn't exist for BridgeTokenV0. Perhaps, migration is not required");`

Reply:
There's also `expect` method for the `Option`

Reply:
👌